src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -0,0 +1,187 @@
+from sentry.models.group import Group
+from sentry.models.organization import Organization, OrganizationStatus
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.release import Release, ReleaseProject
+from sentry.services.hybrid_cloud.user_option import user_option_service
+from sentry.silo import SiloMode
+from sentry.snuba.referrer import Referrer
+from sentry.tasks.base import instrumented_task, retry
+from sentry.tasks.summaries.utils import (
+    ONE_DAY,
+    DailySummaryProjectContext,
+    OrganizationReportContext,
+    fetch_key_error_groups,
+    fetch_key_performance_issue_groups,
+    project_event_counts_for_organization,
+    project_key_errors,
+    project_key_performance_issues,
+    user_project_ownership,
+)
+from sentry.types.activity import ActivityType
+from sentry.types.group import GroupSubStatus
+from sentry.utils.dates import floor_to_utc_day, to_datetime, to_timestamp
+from sentry.utils.outcomes import Outcome
+from sentry.utils.query import RangeQuerySetWrapper
+
+
+# The entry point. This task is scheduled to run every day at 4pm PST.
+@instrumented_task(
+    name="sentry.tasks.daily_summary.schedule_organizations",
+    queue="reports.prepare",
+    max_retries=5,
+    acks_late=True,
+    silo_mode=SiloMode.REGION,
+)
+@retry
+def schedule_organizations(timestamp: float | None = None, duration: int | None = None) -> None:
+    if timestamp is None:
+        # The time that the report was generated
+        timestamp = to_timestamp(floor_to_utc_day(timezone.now()))
+
+    if duration is None:
+        # The total timespan that the task covers
+        duration = ONE_DAY
+
+    organizations = Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+    for organization in RangeQuerySetWrapper(
+        organizations, step=10000, result_value_getter=lambda item: item.id
+    ):
+        if features.has("organizations:daily-summary", organization):
+            user_ids = {
+                user_id
+                for user_id in OrganizationMember.objects.filter(
+                    organization_id=organization.id, teams__projectteam__project__isnull=False
+                ).values_list("user_id", flat=True)
+            }
+
+            # TODO: convert timezones to UTC offsets and group
+            users_by_tz = defaultdict(list)
+            users_with_tz = user_option_service.get_many(
+                filter=dict(user_ids=user_ids, key="timezone")
+            )
+            # if a user has not set a timezone, default to UTC
+            users_without_tz = set(user_ids) - {uo.user_id for uo in users_with_tz}
+            if users_with_tz:
+                users_by_tz["UTC"] = list(users_without_tz)
+            for uo in users_with_tz:
+                users_by_tz[uo.value].append(uo.user_id)
+            for tz in users_by_tz.keys():
+                # Create a celery task per timezone
+                prepare_summary_data(timestamp, duration, organization.id)
```

Comment:
More context on why it's done this way: https://www.notion.so/sentry/Daily-and-Weekly-Issue-Summaries-a69786c9020543d986e349f89a3e78e6?pvs=4#0f7c6677b9444fe085580d334da61d62 the key takeaway being it will barely be a higher number than the number of orgs.

The user id value list is used to get the timezone user options on line 72. 

But you're right I'm not actually using it yet ... I guess I should check for timezones where it's currently 4pm and only generate the reports for those organization's users? I was trying to not overcomplicate it for the MVP so maybe instead I could change it to a user scoped flag, hardcode the PST timezone and just flag some of us in?

Reply:
Ya i guess my assumption was that we'd generate a report _for a user_ who's tz is past 4pm,

but here i don't see anything utilizing the tz, nor do i see how the report relates to a user at all.... so a bit confused what the report we're preparing is :/

Reply:
We'll generate the report if org's user(s) current time is ~4pm, then send the report to the user(s) whose time is ~4pm. (I'm working on adding this now, mentioned in Slack.) This PR doesn't actually do any of the sending yet, but we'll have to check that list again - maybe I could pass the tz / offsets for which to check for the users. I'm not that concerned about that part in this PR.